File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> values;
        unordered_set<string> operators = {"+", "-", "*", "/"};

        for (const string& token : tokens) {
            if (operators.find(token) == operators.end()) {
                values.push(stoi(token));
            } else {
                int right = values.top();
                values.pop();

                int left = values.top();
                values.pop();

                if (token == "+") {
                    values.push(left + right);
                } else if (token == "-") {
                    values.push(left - right);
                } else if (token == "*") {
                    values.push(left * right);
                } else {
                    values.push(left / right);
                }
            }
        }

        return values.top();
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        hash = {')' : '(', '}' : '{', ']' : '['}
        for char in s:
            if char in hash:
                if not stack or stack[-1] != hash[char]:
                    return False
                stack.pop()
            else:
                stack.append(char)
        return not stack
        